### PR TITLE
Change JuliaSyntaxHighlighting UUID

### PR DESCRIPTION
**Pull request by tecosaur**
TLDR; I made a silly mistake a while back, and this seems like the least bad option.

-----

Unfortunately during development of the backported version of this stdlib, to avoid potential sysimage issues the UUID was temporarilly changed and I forgot to change it back.

This mistake lead to the backport/compat verison of the package having a different UUID from the in-tree stdlib. Beyond this being an embarassing mistake, it's also a pain because it means that we now have incorrect/invalid manifests.

A small mercy is that 1.12 hasn't been released yet, so the discrepancy currently only affects people using development Julia versions. Changing the UUID here will require these people to regenerate their manifests, but this seems like the lesser of two evils.

Appologies for the hassle all.

### CODE REVIEW

File: stdlib/Manifest.toml [hunk 1]
```diff
@@ -67,7 +67,7 @@ version = "1.11.0"
 
 [[deps.JuliaSyntaxHighlighting]]
 deps = ["StyledStrings"]
-uuid = "dc6e5ff7-fb65-4e79-a425-ec3bc9c03011"
+uuid = "ac6e5ff7-fb65-4e79-a425-ec3bc9c03011"
```
Comment (inkydragon) [inline stdlib/Manifest.toml:70]: I'm a little confused, it seems that the UUID of `JuliaSyntaxHighlighting.jl` is specified as `dc6e5ff7-fb65-4e79-a425-ec3bc9c03011` from the first commit.
So the current UUID seems to be correct?

https://github.com/JuliaLang/JuliaSyntaxHighlighting.jl/blame/19bd57b89c648592155156049addf67e0638eab1/Project.toml#L2

Comment (tecosaur) [reply stdlib/Manifest.toml:70]: Yea, so the mistake at the heart of this is:

> development of the backported version of this stdlib, to avoid potential sysimage issues the UUID was temporarilly changed and I forgot to change it back.

The changed UUID (`ac6e...`) is what was registered in General as the backport/compat version: https://github.com/JuliaRegistries/General/blob/master/J/JuliaSyntaxHighlighting/Package.toml#L2

Comment (inkydragon) [reply stdlib/Manifest.toml:70]: I think JuliaSyntaxHighlighting.jl also needs a pr to change the UUID and the two prs might want to be merged at the same time?

**Comment by inkydragon** [comment]
@tecosaur Any Update?
Perhaps we should include this in 1.12.

**Comment by tecosaur** [comment]
We should, and I'll do exactly that — I've just been travelling recently.

**Comment by inkydragon** [comment]
Kindly ping @tecosaur 

I guess we'll move to v1.13 this week #56970

File: stdlib/Manifest.toml (hunk 1 above)
Comment (tecosaur) [reply stdlib/Manifest.toml:70]: Oh, I just went to do this and noticed I already did it, that's why the `sha1` is updated. Merging this PR will use the version of `JuliaSyntaxHighlighting` with the updated UUID (otherwise the tests would have been failing).

Comment (tecosaur) [reply stdlib/Manifest.toml:70]: (the commit is just in a branch, not a PR)

Comment (inkydragon) [reply stdlib/Manifest.toml:70]: The changes are located on the `julia1-compat` branch:
xref: https://github.com/JuliaLang/JuliaSyntaxHighlighting.jl/blob/julia1-compat/Project.toml#L2

File: stdlib/JuliaSyntaxHighlighting.version [hunk 2]
```diff
@@ -1,4 +1,4 @@
 JULIASYNTAXHIGHLIGHTING_BRANCH = main
-JULIASYNTAXHIGHLIGHTING_SHA1 = 19bd57b89c648592155156049addf67e0638eab1
+JULIASYNTAXHIGHLIGHTING_SHA1 = eb8097c5f1dbfca80b5e6664af031ff1fe0904af
```
Comment (inkydragon) [inline stdlib/JuliaSyntaxHighlighting.version:2]: When I was checking this commit, github gave a warning:
> This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

commit: https://github.com/JuliaLang/JuliaSyntaxHighlighting.jl/commit/eb8097c5f1dbfca80b5e6664af031ff1fe0904af

The change should be pushed into one of the `JuliaLang/JuliaSyntaxHighlighting.jl` branches.
Maybe `julia1-compat`?

**Review by inkydragon — APPROVED** [review]
LGTM

A quick recap:
We have registered the wrong UUID (`ac6e5ff7-`) for JuliaSyntaxHighlighting.jl in General.

```toml
name = "JuliaSyntaxHighlighting"
uuid = "ac6e5ff7-fb65-4e79-a425-ec3bc9c03011"
repo = "https://github.com/JuliaLang/JuliaSyntaxHighlighting.jl.git"
```
https://github.com/JuliaRegistries/General/blob/48cba3e8f391da43e7df47264f333c210ced7192/J/JuliaSyntaxHighlighting/Package.toml#L2

To minimize the impact, we chose to make the UUID registered in General the correct one.

So we need to update the UUID in Project.toml of JuliaLang/JuliaSyntaxHighlighting.jl.

`dc6e5ff7-*` => `ac6e5ff7-*`

```toml
name = "JuliaSyntaxHighlighting"
uuid = "dc6e5ff7-fb65-4e79-a425-ec3bc9c03011"
```
https://github.com/JuliaLang/JuliaSyntaxHighlighting.jl/blob/19bd57b89c648592155156049addf67e0638eab1/Project.toml#L1C1-L2C46

-----

I'll update this branch to make sure everything works on the latest master branch.

File: stdlib/JuliaSyntaxHighlighting.version (hunk 2 above)
Comment (tecosaur) [reply stdlib/JuliaSyntaxHighlighting.version:2]: Ah yep, since this looks good to merge I think it's not presumptuous to just put that commit in the `main` branch now. I'll do so when I next sit down in front of my computer.

**Comment by inkydragon** [comment]
I'm going to merge this pr in about a day or two.

Add the `merger me` tag to attract any potential reviewers.

**Comment by DilumAluthge** [comment]
I think we need to get this PR in before 1.12 is branched. It would be good if someone else could take a look at this and make sure it looks correct/won't break anything.

@KristofferC Could you review this?

**Comment by felixcremer** [comment]
Is there an easy way to regenerate the current manifest file from within julia?

**Comment by IanButterworth** [comment]
@felixcremer not that I'm aware of.

You have two options:
1. @tecosaur mentioned that you can switch the UUID out with
```
sed -i 's/dc6e5ff7-fb65-4e79-a425-ec3bc9c03011/ac6e5ff7-fb65-4e79-a425-ec3bc9c03011/' ~/.julia/environments/v1.12/Manifest.toml
```

2. Delete the manifest file and `pkg> resolve` but that loses versioning state
